core/src/main/scala/Graph.scala

```diff
@@ -903,6 +903,30 @@ case class Graph[N,A,B](rep: GraphRep[N,A,B]) {
     getLPath(t, lbft(s))
 
   /**
+   * Cheapest path from vertex `s` to vertex `t` under the cost function `costFkt` with labels
+   * @group bfs   
+   */
+  def cheapestPath[C](s: N, t: N, costFkt: (LNode[N,A],B,LNode[N,A]) => C)(implicit num: Numeric[C]): Option[LPath[N,B]] = {
+    require(contains(s))
+    require(contains(t))
```

Comment:
Could we return `None` here, or is it important to distinguish a missing vertex from an empty searchtree?  The exceptions make me a bit uncomfortable.

Reply:
Yeah, ideally we don't throw exceptions but instead codify things into the type system :-)

Reply:
Rewrite all of this in idris and I will happily encode that in the signature :)

Reply:
Well, until someone publishes quividris, my question remains. :)  Would a `None` be an acceptable result, or do we need a third value to return?

Reply:
Sorry for not answering this sooner. I think that would be perfectly fine.